client/my-sites/plans-features-main/hooks/use-check-plan-availability-for-purchase.ts

```diff
@@ -21,8 +22,9 @@ type PlanAvailabilityForPurchase = {
  */
 const useCheckPlanAvailabilityForPurchase = ( {
 	planSlugs,
+	siteId,
 }: Props ): PlanAvailabilityForPurchase => {
-	const selectedSiteId = useSelector( getSelectedSiteId );
+	const selectedSiteId = useSelector( getSelectedSiteId ) || siteId;
```

Comment:
If we move forward with this approach, we'll want to include a comment about why `siteId` is also needed here, when, presumably, selectedSiteId would be enough.